h/views/accounts.py

```diff
@@ -572,6 +576,63 @@ def _template_data(self, js_config=None):
         return {"js_config": js_config}
 
 
+@view_config(
+    route_name="account_identity",
+    is_authenticated=True,
+    request_method="POST",
+    require_csrf=True,
+    request_param=("provider", "provider_unique_id"),
+)
+def delete_identity(request):
+    user = request.user
+    db = request.db
+    flash = request.session.flash
+    params = request.params
+    route_url = request.route_url
+
+    try:
+        given_provider = IdentityProvider[params["provider"].upper()]
+    except KeyError as err:
+        raise httpexceptions.HTTPNotFound from err
```

Comment:
Is the expectation here is that this should never be triggered so you don't need a "nice" error message for it?

Reply:
Yeah. This should never be hit by users using our own UI, because of course our UI should never contain forms with unknown providers in them. But from the point of view of the backend view, technically someone could send us a request with an unknown provider, so we have to do something other than just crashing.